File: Top_100_Liked_Questions/56._Merge_Intervals/solution.py
class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        if not intervals:
            return []
        
        intervals.sort(key=lambda x: x[0])
        ans = [intervals[0]]
        
        for s,e in intervals:
            if s > ans[-1][1]:
                ans.append([s,e])
            else:
                ans[-1][1] = max(e, ans[-1][1])
        
        return ans

File: Top_100_Liked_Questions/56._Merge_Intervals/solution.java
class Solution {
    public int[][] merge(int[][] intervals) {
        if (intervals.length <= 1)
            return intervals;

        Arrays.sort(intervals, (a, b) -> {
            if (a[0] != b[0])
                return a[0] - b[0];
            else
                return a[1] - b[1];
        });

        int count = 1;
        int i = 0;
        while (i < intervals.length-1) {
            if (intervals[i+1][0] <= intervals[i][1]) {
                intervals[i+1][0] = intervals[i][0];
                intervals[i+1][1] = Math.max(intervals[i][1], intervals[i+1][1]);
                intervals[i][0] = -1;
            }
            else
                count++;

            i++;
        }

        int[][] ans = new int[count][2];
        int k=0;
        for (int j=0; j<intervals.length; j++) {
            if (intervals[j][0] >= 0) {
                ans[k][0] = intervals[j][0];
                ans[k][1] = intervals[j][1];
                k++;
            }
        }

        return ans;
    }
}